code/modules/antagonists/wizard/equipment/soulstone.dm

```diff
@@ -10,12 +10,15 @@
 	desc = "A fragment of the legendary treasure known simply as the 'Soul Stone'. The shard still flickers with a fraction of the full artefact's power."
 	w_class = WEIGHT_CLASS_TINY
 	slot_flags = ITEM_SLOT_BELT
-	var/usability = FALSE
-
 	var/old_shard = FALSE
 	var/spent = FALSE
-///this controlls the color of the soulstone as well as restrictions for who can use it. THEME_CULT is red and is the default of cultist THEME_WIZARD is purple and is the default of wizard and THEME_HOLY is for purified soul stone
+	/// This controls the color of the soulstone as well as restrictions for who can use it. THEME_CULT is red and is the default of cultist THEME_WIZARD is purple and is the default of wizard and THEME_HOLY is for purified soul stone
 	var/theme = THEME_CULT
+	/// Role check, if any needed
+	var/role_to_check = /datum/antagonist/cult
+
+/obj/item/soulstone/proc/role_check(mob/who)
+	return who.mind && who.mind.has_antag_datum(role_to_check, TRUE)
```

Comment:
```suggestion
#define SOULSTONE_ROLE_CHECK(role_to_check, mob_to_check) ( isnull(role_to_check) ? TRUE : mob_to_check.mind?.has_antag_datum(role_to_check, TRUE) ? TRUE : FALSE )
```
This should probably be closer to what you want.
Converted to boolean because `has_antag_datum()` returns a reference or `null`.